### PR TITLE
Add hide intent to cancelled downloads

### PR DESCRIPTION
**Pull request by ronocod**
In a situation where we had 2 downloads, one in progress and one cancelled, swiping away the cancelled one didn't hide it permanently.

This was because we weren't adding the hide Intent to cancelled notifications.

This PR adds the hide intent to cancelled notifications and adds some boy scout refactoring.

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/213/
Test PASSed.

File: library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java [hunk 1]
```diff
@@ -96,34 +97,30 @@ private void buildActionIntents(String tag, int type, Collection<DownloadBatch>
...
-            DownloadBatch batch = cluster.iterator().next();
             Intent cancelIntent = new Intent(Constants.ACTION_CANCEL, null, context, DownloadReceiver.class);
             cancelIntent.putExtra(DownloadReceiver.EXTRA_BATCH_ID, batch.getBatchId());
             PendingIntent pendingCancelIntent = PendingIntent.getBroadcast(context, 0, cancelIntent, PendingIntent.FLAG_UPDATE_CURRENT);
             builder.addAction(R.drawable.dl__ic_action_cancel, context.getString(R.string.dl__cancel), pendingCancelIntent);
 
-        } else if (type == DownloadNotifier.TYPE_SUCCESS) {
-            DownloadBatch batch = cluster.iterator().next();
+        } else if (type == DownloadNotifier.TYPE_SUCCESS || type == DownloadNotifier.TYPE_CANCELLED) {
             // TODO: Decide how we handle notification clicks
```
Comment (friedger) [inline library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java]: What is left for this todo?

**Comment by friedger** [comment]
Looks good

File: library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java (hunk 1 above)
Comment (ronocod) [inline library/src/main/java/com/novoda/downloadmanager/lib/NotificationDisplayer.java]: I think it's related to the `clickIntent` which is now on Line `119`. If there's no extra default behaviour that we want to add here then the comment can probably be removed

**Pull request merged**